vault/mount.go

```diff
@@ -1021,15 +1019,10 @@ func (c *Core) setupMounts(ctx context.Context) error {
 		var backend logical.Backend
 		// Create the new backend
 		sysView := c.mountEntrySysView(entry)
-		// Set up conf to pass in plugin_name
-		conf := make(map[string]string)
-		if entry.Config.PluginName != "" {
```

Comment:
I think there needs to be upgrade logic somewhere in `loadMounts` (and for the auth equivalent) to upgrade the mount table, since an existing mount table may have the plugin name and `entry.Type` as `"plugin"` stored. I haven't ran this, but the current setupMount changes might render any existing plugin mounts to not get mounted properly.

Reply:
Hah, I guess same comment as https://github.com/hashicorp/vault/pull/5536#discussion_r229899429. Seems that you're working on fixing it.

Reply:
This is a very good idea. Upon discussion, it turned out that this wouldn't work with replication secondaries due to the replication logic and the fact that we need to support higher versions on secondaries, so we chose an alternate route.